matrix/matrix.py

```diff
@@ -0,0 +1,184 @@
+        return repr(self._matrix)
+
+    @property
+    def tuples(self):
+        """Return the matrix instance as a tuple of tuples."""
+        return self._matrix
+
+    def _matrix_by_matrix(self, other, operation):
+        """Calculate operation on two matrices.
+
+        Args:
+            operation (function): the function of the operator.
+            other (Matrix): the other matrix to operate on.
+
+        Returns:
+            Matrix: the result of the operation.
+
+        Note:
+            Boolean operations supports un-squared matrices but
+            mathematical operations don't.
+        """
+        if len(other.tuples) != len(self.tuples) and not any(
+                [operation == operator.eq, operation == operator.ne]):
+            raise ValueError("Matrices length don't match")
+
+        calculated_matrix = []
+
+        for i in range(len(self.tuples)):
+
+            calculated_line = []
+            for j in range(len(self.tuples)):
+                calculated_line.append(
+                    operation(self.tuples[i][j], other.tuples[i][j]))
+
+            calculated_matrix.append(tuple(calculated_line))
+
+        return Matrix(*calculated_matrix)
+
+    def _by_scalar(self, scalar, operation):
+        """Calculate operation on matrix by scalar.
+
+        Args:
+            operation (function): the function of the operator.
+            scalar (number): the scalar to operate by.
+
+        Returns:
+            Matrix: the result of the operation.
+
+        Note:
+            Scalar must be a number. Operations with scalars are only * and /.
+        """
+        calculated_matrix = []
+
+        for i in range(len(self.tuples)):
+
+            calculated_line = []
+            for j in range(len(self.tuples)):
+                calculated_line.append(operation(self.tuples[i][j], scalar))
+
+            calculated_matrix.append(tuple(calculated_line))
+
+        return Matrix(*calculated_matrix)
+
+    def __add__(self, other):
+        """Add two matrices."""
+        return self._matrix_by_matrix(other, operator.add)
+
+    def __sub__(self, other):
+        """Subtract two matrices."""
+        return self._matrix_by_matrix(other, operator.sub)
+
+    def __mul__(self, other):
+        """Multiply two matrices or scalar by matrix."""
+        if is_number(other):
+            return self._by_scalar(other, operator.mul)
+
+        return self._matrix_by_matrix(other, operator.mul)
+
+    def __rmul__(self, other):
+        """Define to handle reverse multiplication (10 * matrix) as regular
+        multiplication."""
+        return self.__mul__(other)
+
+    def __truediv__(self, other):
+        """Divide two matrices or scalar by matrix. Returns floats if
+        needed."""
+        if not is_number(other):
+            raise TypeError(
+                f'Matrices division not supported on {type(other)}')
+
+        return self._by_scalar(other, operator.truediv)
+
+    def __floordiv__(self, other):
```

Comment:
Floordir should either be deleted or perform the same as the default floor division, not being exactly like the regular div